### ISSUE TITLE
 Invalid C++ declaration: Expected identifier in nested name.

### ISSUE THREAD

**Issue by cliffburdick** [sphinx-doc/sphinx#10016]
### Describe the bug

When creating documentation for this project Sphinx says:

```
Exception occurred:
  File "/usr/local/lib/python3.8/dist-packages/sphinx/util/cfamily.py", line 275, in fail
    raise self._make_multi_error(errors, '')
sphinx.util.cfamily.DefinitionError: Invalid C++ declaration: Expected identifier in nested name. [error at 8]
  typename...
  --------^
```

It seems to occur on a variadic template. 

### How to Reproduce

```
$ git clone https://github.com/NVIDIA/MatX.git
$ cd MatX
$ (with cmake, doxygen/breathe/sphinx installed"
$ cmake .. -DBUILD_DOCS=ON
$ make

```

### Expected behavior

Documentation generates successfully

### Your project

https://github.com/NVIDIA/MatX.git

### OS

Ubuntu 20.04

### Python version

3.8

### Sphinx version

4.3.2

### Sphinx extensions

extensions = ['breathe', 'sphinx.ext.autodoc', 'sphinx_rtd_theme', 'sphinx.ext.mathjax']

**Comment by jakobandersen** [comment]
Which branch is this on? With the default I get a missing ``CMakeLists.txt`` file error for the ``docs/`` folder.
Some more information that would be useful:
- What is the full backtrace?
- Which version of Breathe are you using? My first guess is that you have hit the problem fixed by https://github.com/michaeljones/breathe/pull/711, which should be in Breathe 4.31.

**Comment by cliffburdick** [comment]
Hi @jakobandersen, thanks for responding. You're right that I forgot to commit some recent changes, but now ``main`` should have the right directories. I will try the latest Breathe and get back to you

**Comment by cliffburdick** [comment]
Hi @jakobandersen confirmed this is fixed in 4.31. Thanks!

**Comment by jakobandersen** [comment]
Great, thanks!
